coderd/workspaceagents.go

```diff
@@ -1891,12 +1896,22 @@ func (api *API) watchWorkspaceAgentMetadata(rw http.ResponseWriter, r *http.Requ
 	}
 	// Prevent handler from returning until the sender is closed.
 	defer func() {
+		cancel()
 		<-sseSenderClosed
 	}()
+	// Synchronize cancellation from SSE -> context, this lets us simplify the
+	// cancellation logic.
+	go func() {
+		select {
+		case <-ctx.Done():
+		case <-sseSenderClosed:
+			cancel()
+		}
+	}()
```

Comment:
These changes were added so that we can be sure we're not blocking on `<-sseSenderClosed`.